mllib/src/main/scala/org/apache/spark/ml/feature/HashingTF.scala

```diff
@@ -132,27 +141,39 @@ class HashingTF @Since("1.4.0") (@Since("1.4.0") override val uid: String)
   override def toString: String = {
     s"HashingTF: uid=$uid, binary=${$(binary)}, numFeatures=${$(numFeatures)}"
   }
+
+  @Since("3.0.0")
+  override def save(path: String): Unit = {
+    require(hashFuncVersion == HashingTF.SPARK_3_MURMUR3_HASH,
+      "Cannot save model which is loaded from lower version spark saved model.")
```

Comment:
Tell users how to solve the issue:
* Use old versions of Spark.
* Re-train the pipeline.